catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/formulaeditor/FormulaEditorKeyboardTest.java

```diff
@@ -156,6 +160,32 @@ public void sensorFormulaTest() {
 		onView(withId(R.id.brick_set_variable_edit_text))
 				.check(matches(withText(activity.getString(R.string.formula_editor_sensor_x_acceleration) + " + 2 ")));
 	}
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void textFunctionTest() {
+		Activity activity = baseActivityTestRule.getActivity();
+		if (activity == null || activity.isFinishing()) {
+			Log.e("Activity error", "Formula Editor Keyboard Text Activity is not Found.");
+			return;
+		}
+		onView(withId(R.id.brick_set_variable_edit_text)).perform(click());
+		onView(withId(R.id.formula_editor_keyboard_text)).perform(click());
+		onView(withText("length('hello world')")).check(matches(isDisplayed()));
+		onView(withText("letter(1,'hello world')")).check(matches(isDisplayed()));
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void listFunctionTest() {
+		Activity activity = baseActivityTestRule.getActivity();
+		if(activity == null || activity.isFinishing()) {
```

Comment:
```suggestion
		if (activity == null || activity.isFinishing()) {
```
https://jenkins.catrob.at/job/Catroid/job/PR-3950/7/analysis/ This will fix the static analysis warning